File: src/main/java/dev/losterixx/sCore/paper/features/warps/ListWarpsCommand.java
package dev.losterixx.sCore.paper.features.warps;

import dev.dejvokep.boostedyaml.YamlDocument;
import dev.dejvokep.boostedyaml.block.implementation.Section;
import dev.losterixx.sCore.paper.PaperMain;
import dev.losterixx.sCore.paper.utils.ConfigManager;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.Sound;
import org.bukkit.World;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;
import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.List;

public class ListWarpsCommand implements CommandExecutor {

    private MiniMessage mm = PaperMain.mm;
    private PaperMain main = PaperMain.getInstance();
    private ConfigManager configManager = main.getConfigManager();
    private YamlDocument getConfig() { return configManager.getConfig("config"); }
    private YamlDocument getMessages() { return configManager.getConfig("messages"); }
    private YamlDocument getData() { return configManager.getConfig("data"); }
    private Component getPrefix() { return mm.deserialize(getConfig().getString("prefix")); }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {

        if (!(sender instanceof Player)) {
            sender.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("general.noPlayer"))));
            return false;
        }

        if (!sender.hasPermission("sCore.command.listwarps")) {
            sender.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("general.noPerms"))));
            return false;
        }

        if (args.length != 0) {
            sender.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("commands.listwarps.usage"))));
            return false;
        }

        Player player = (Player) sender;

        Section warpsSection = getData().getSection("warps");

        if (warpsSection == null) {
            player.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("commands.listwarps.noWarps"))));
            return false;
        }

        player.sendMessage(mm.deserialize(getMessages().getString("commands.listwarps.header")));
        for (Object warpName : warpsSection.getKeys()) {
            if (!(warpName instanceof String)) continue;
            String key = (String) warpName;
            player.sendMessage(mm.deserialize(getMessages().getString("commands.listwarps.entry").replaceAll("%warp%", key)));
        }
        player.sendMessage(mm.deserialize(getMessages().getString("commands.listwarps.footer")));

        return false;
    }

}

File: src/main/kotlin/dev/losterixx/sCore/features/warps/ListWarpsCommand.kt
package dev.losterixx.sCore.features.warps

import dev.losterixx.sCore.Main
import dev.losterixx.sCore.utils.ConfigManager
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class ListWarpsCommand : CommandExecutor {

    private val mm = Main.miniMessage
    private val main = Main.instance
    private fun getConfig() = ConfigManager.getConfig("config")
    private fun getMessages() = ConfigManager.getConfig(getConfig().getString("langFile", "english"))
    private fun getData() = ConfigManager.getConfig("data")
    private fun getPrefix() = getConfig().getString("prefix") ?: Main.DEFAULT_PREFIX

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {

        if (sender !is Player) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("general.noPlayer")))
            return false
        }

        if (!sender.hasPermission("sCore.command.listwarps")) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("general.noPerms")))
            return false
        }

        if (args.isNotEmpty()) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("commands.listwarps.usage")))
            return false
        }

        val warpsSection = getData().getSection("warps")
        if (warpsSection == null || warpsSection.getRoutesAsStrings(false).isEmpty()) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("commands.listwarps.noWarps")))
            return false
        }

        sender.sendMessage(mm.deserialize(getMessages().getString("commands.listwarps.warpsHeader")
            .replace("%prefix%", getPrefix())))

        for (warp in warpsSection.getRoutesAsStrings(false)) {
            val line = getMessages().getString("commands.listwarps.warpsElement")!!
                .replace("%prefix%", getPrefix())
                .replace("%warp%", warp)
            sender.sendMessage(mm.deserialize(line))
        }

        sender.sendMessage(mm.deserialize(getMessages().getString("commands.listwarps.warpsFooter")
            .replace("%prefix%", getPrefix())))

        return false
    }

}